ckanext/canada/tests/test_hospitalityq.py

```diff
@@ -42,6 +42,51 @@ def test_blank(self):
         assert 'ref_number' in err['key'][0]
 
 
+    def test_vendor_format(self):
+        record = get_chromo('hospitalityq')['examples']['record']
+
+        # tests some incorrect formats
+        incorrect_formats = [
+            'Starbucks ;',
+            'Tim Hortons & Starbucks ;   ',
+            'Les Impertinentes ; ; Les Street Monkeys',
+            'Starbucks, Montanas ; ',
+        ]
+        for incorrect_format in incorrect_formats:
+            record['vendor_other_en'] = incorrect_format
+            record['vendor_other_fr'] = incorrect_format
+            with pytest.raises(ValidationError) as ve:
+                self.lc.action.datastore_upsert(
+                    resource_id=self.resource_id,
+                    records=[record])
+            err = ve.value.error_dict['records'][0]
+            expected = {
+                'vendor_other_en': ["Invalid format for multiple commercial establishments or vendors. Use {Vendor Name};{Vendor 2 Name} (e.g. Les Impertinentes;Les Street Monkeys)"],
+                'vendor_other_fr': ["Invalid format for multiple commercial establishments or vendors. Use {Vendor Name};{Vendor 2 Name} (e.g. Les Impertinentes;Les Street Monkeys)"],
+            }
+            for k in set(err) | set(expected):
+                assert k in err
+                assert err[k] == expected[k]
```

Comment:
instead of raising an error would it be better to strip the empty vendors?

Reply:
I just couldn't get the regex the way I wanted it to, so that is why I did an inner regex to raise the error as well